src/worker.cpp

```diff
@@ -651,12 +656,56 @@ bool Worker::isCPUIOGoodOnly(const std::string& pgKeyword)
     return false;
 }
 
-void Worker::primeInventory(const types::IPZVpdMap& ipzVpdMap,
-                            types::ObjectMap primeObjects)
+void Worker::primeInventory(const std::string& i_vpdFilePath,
+                            types::ObjectMap& io_objectInterfaceMap)
 {
-    // TODO implementation
-    (void)ipzVpdMap;
-    (void)primeObjects;
+    logging::logMessage("primeInventory called for FRU " + i_vpdFilePath);
+
+    if (i_vpdFilePath.empty())
+    {
+        throw std::runtime_error("Empty VPD file path given");
+    }
+
+    if (!m_parsedJson.empty())
```

Comment:
I would suggest check for json and throw from here. get rid of else part.
This will reduce the level of indentation and will make the code more readable.

Reply:
updated.